lib/archethic/contracts/interpreter/library.ex

```diff
@@ -149,4 +153,27 @@ defmodule ArchEthic.Contracts.Interpreter.Library do
   def size(binary) when is_binary(binary), do: byte_size(binary)
   def size(list) when is_list(list), do: length(list)
   def size(map) when is_map(map), do: map_size(map)
+
+  @doc """
+  Get the genesis address of the chain
+
+  """
+  @spec get_genesis_address(binary()) ::
+          binary() | {:error, :network_issue} | {:error, :enoaddress}
+  def get_genesis_address(address) do
+    address = Base.decode16!(address)
+
+    with [node | _rest] <- P2P.available_nodes(),
+         public_key_request <- %GetFirstPublicKey{address: address},
+         {:ok, %FirstPublicKey{public_key: key}} <- P2P.send_message(node, public_key_request) do
+      {pub, _priv} = Crypto.derive_keypair(key, 0)
+      Crypto.derive_address(pub)
+    else
+      [] ->
+        {:error, :network_issue}
+
+      {:ok, %NotFound{}} ->
```

Comment:
As mentioned before, we have to change the implementation of the P2P message to not return NotFound, but the public key passed, if the previous public key does not exist.
Because this function should return a string, we have to design it with this in mind.
So I think you can return address, in case of not found, until the fix of the P2P message is done.